applications/nrf_desktop/src/modules/watchdog.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/watchdog.h>
+
+#define MODULE watchdog
+#include "module_state_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_WATCHDOG_LOG_LEVEL);
+
+
+#if defined(CONFIG_WDT_0_NAME)
+#define WATCHDOG_DEV_NAME CONFIG_WDT_0_NAME
+#elif defined(DT_WDT_0_NAME)
+#define WATCHDOG_DEV_NAME DT_WDT_0_NAME
+#else
+#error "Error!!! Watchdog device name not defined - verify configuration."
+#endif
+#define WDT_FEED_WORKER_DELAY_MS 2000
```

Comment:
can this be calculated from configured delay?
e.g. `delay/3`?

Reply:
yes it can be evaluated - change done